pkg/collector/check/context.go

```diff
@@ -0,0 +1,56 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package check
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/comp/metadata/inventorychecks"
+)
+
+// checkContext holds a list of reference to different components used by Go and Python checks.
+//
+// This is a temporary solution until checks are components themselves and can request dependencies through FX.
+//
+// This also allows Go function exported to CPython to recover there reference to different components when coming out
```

Comment:
`// This also allows Go function`. Is it a temporary solution, `also` let me think it is not the case?